File: backend/app/routers/feedback.py
from fastapi import APIRouter, Depends

from app.auth import require_roles
from app.db import get_db
from app.models import FeedbackCreateRequest
from app.repositories import create_feedback_entry, list_feedback_entries, summarize_feedback_entries

router = APIRouter(prefix="/api/feedback", tags=["feedback"])


@router.post("")
async def post_feedback(payload: FeedbackCreateRequest, user: dict = Depends(require_roles("admin", "member", "viewer"))) -> dict:
    with get_db() as db:
        feedback = create_feedback_entry(
            db,
            int(user["id"]),
            str(user["role"]),
            payload.rating,
            payload.comment,
            payload.feature_name,
        )
        summary = summarize_feedback_entries(db)
    return {"feedback": feedback, "summary": summary}


@router.get("")
async def get_feedback(_: dict = Depends(require_roles("admin"))) -> dict:
    with get_db() as db:
        return {
            "feedback": list_feedback_entries(db, 200),
            "summary": summarize_feedback_entries(db),
        }

File: frontend/api/feedback.ts
import { fetchJson } from "@/api/client";
import type { FeedbackEntry, FeedbackRating, FeedbackSummary } from "@/types/app";

export function getFeedback(): Promise<{ feedback: FeedbackEntry[]; summary: FeedbackSummary }> {
  return fetchJson("/api/feedback");
}

export function submitFeedback(payload: {
  rating: FeedbackRating;
  comment: string;
  feature_name: string;
}): Promise<{ feedback: FeedbackEntry; summary: FeedbackSummary }> {
  return fetchJson("/api/feedback", {
    method: "POST",
    body: JSON.stringify(payload)
  });
}
